go/libraries/doltcore/doltdb/doltdb.go

```diff
@@ -186,18 +188,72 @@ func (ddb *DoltDB) WriteEmptyRepoWithCommitTimeAndDefaultBranch(
 		return err
 	}
 
-	cm, _ := datas.NewCommitMetaWithUserTS(name, email, "Initialize data repository", t)
+	// --fun is a hidden command line option that rerolls the initial commit until
+	// it starts with "d0lt" or similar. We achieve this by replacing characters
+	// in the commit description with Cyrillic equivalents, then by decrementing
+	// the timestamp.
+	descriptionReplacementCandidates := [][]rune{
+		{'I', '\u0406'},
+		{'i', '\u0456'},
+		{'i', '\u0456'},
+		{'a', '\u0430'},
+		{'i', '\u0456'},
+		{'e', '\u0435'},
+		{'a', '\u0430'},
+		{'a', '\u0430'},
+		{'e', '\u0435'},
+		{'o', '\u043e'},
+		{'i', '\u0456'},
+		{'o', '\u043e'},
+	}
+	attempt := 0
+	var firstCommit *datas.Commit
+	for {
+		generateInitialCommitDesc := func(attempt int) string {
+			// "Initialize data repository", with characters that could be Cyrillic replaced.
+			descFmt := "%cn%ct%c%cl%cz%c d%ct%c r%cp%cs%ct%cry"
+			choices := []any{}
```

Comment:
(minor go tip) Mostly just sharing this Go convention we use... the perf impact is very very small here, but it's still a good pattern to know. If we know the length of the slice (len of descriptionReplacementCandidates here) then we prefer to preallocate the capacity we need and not have to worry about the slice auto resizing while we're adding to it. Note that the middle param (`0`) is really important if you're using append to build the slice (but not what you want if you're going to walk through the index and set positions manually). 

```suggestion
			choices := make([]rune, 0, len(descriptionReplacementCandidates))
```

Reply:
I refactored it. Note that the array has to be of type []any, not []rune, because go arrays are invariant and Sprintf expects an []any.